backend/src/apiserver/model/pipeline_version.go

```diff
@@ -29,23 +29,28 @@ const (
 )
 
 type PipelineVersion struct {
-	UUID           string `gorm:"column:UUID; not null; primary_key;"`
+	UUID           string `gorm:"column:UUID; not null; primaryKey;"`
 	CreatedAtInSec int64  `gorm:"column:CreatedAtInSec; not null; index;"`
-	Name           string `gorm:"column:Name; not null; unique_index:idx_pipelineid_name;"`
-	DisplayName    string `gorm:"column:DisplayName; not null"`
+	// Explicitly specify type:varchar(255) to ensure MySQL can create index on this column.
+	// In GORM v1, omitting type still allowed index creation due to default behavior.
+	// However, GORM v2 requires explicit type declaration for indexed string fields,
+	// otherwise it may default to longtext, which is not indexable in MySQL.
```

Comment:
```suggestion
	// Explicitly specify type:varchar(255) to ensure MySQL can create index on this column.
	// GORM requires explicit type declaration for indexed string fields,
	// otherwise it may default to longtext, which is not indexable in MySQL.
```

as before, the removed comments here are better to include as commit message

Reply:
> omitting type still allowed index creation due to default behavior.

can you elaborate, are you saying in gorm v1 users could create a name > 255 char and it was indexable, but now that's not possible in gorm v2? 

just trying to understand if we are making an api breaking change here by adding a limitation on pipeline name

Reply:
Thanks — I did some more digging and realized the core issue here is really MySQL’s indexing limit, independent of how GORM behaves.

MySQL imposes a 767-byte limit on [indexed columns](https://dev.mysql.com/doc/refman/8.4/en/innodb-limits.html) in many default configurations. With `utf8mb4`, each character can take up to 4 bytes, so the safest length for indexable string columns is `varchar(191)` — anything larger risks hitting the byte limit and causing a schema error when creating an index.

So I’m switching indexed string fields to `varchar(191)` to stay within the safe range, regardless of the ORM version or database settings. Let me know if you have any concerns with this direction.